app/controllers/students_controller.rb

```diff
@@ -144,29 +148,52 @@ def student_feed(student, restricted_notes: false)
     }
   end
 
-  def set_up_sped_data
-    @student = Student.find(params[:id])
-    @current_educator = current_educator
-    @url = root_url.chomp('/') + request.path
+  def school_year_id_range(from_date, to_date) 
+    # Find the years represented by the dates
 
-    # Calculate the current and prior school year ids for use with report data
-    # only displaying 2 years of data
-    current_school_year_id = DateToSchoolYear.new(Date.today).convert.id
-    prior_school_year_id = DateToSchoolYear.new(Date.today.ago(1.years)).convert.id
+    #start with to_date and work backward each year until < from_date
+    current_date = to_date
+    school_year_ids = []
 
+    while current_date > from_date do
+      school_year_ids.push(DateToSchoolYear.new(current_date).convert.id)
 
-    # Load all event notes that are NOT restricted for the student
-    @event_notes = @student.event_notes.where(:is_restricted => false)
+      current_date = current_date - 1.year
+    end
 
-    # Load all services for the student
-    @services = @student.services.includes(:discontinued_services)
+    return school_year_ids
 
-    # Load last 2 student schools years for absences, tardies, and discipline incidents
-    @student_school_years = @student.student_school_years.includes(:absences).includes(:tardies).includes(:discipline_incidents).where(school_year_id: [prior_school_year_id, current_school_year_id])
+  end
+
+  def set_up_student_report_data
+    @student = Student.find(params[:id])
+    @current_educator = current_educator
+    @url = root_url.chomp('/') + request.path
```

Comment:
I realize that's a holdover from @erose's initial take on the report. Initially the url was included in the footer of the report. I removed it to save space.

So we can remove the variable as well.